### PR TITLE
BUG: Fix small issues found using valgrind

### PR DESCRIPTION
**Pull request by seberg**
(And one found by eye looking.)

---

More issues are most likely coming, since my test run is only 50% done (and I may need to restart to be sure I didn't miss anything in the first half, since repeated errors are distracting).  Further, I will try `pytest-leaks` later and that is more sensitive (but less helpful).

But also fine to put this in, and I will open a new PR.

### CODE REVIEW

**Comment by charris** [comment]
I'll leave this up, make an rc2 release, and we can put these small fixes in before the final.

**Comment by seberg** [comment]
This seems to be it for valgrind (I may run again to be sure, but all remaining things are either repetitions of the same errors or seem to be some imports, which often leak a bit).
(pytest-leaks check upcoming in the next few days, probably)

File: numpy/core/src/umath/dispatching.c [hunk 1]
```diff
@@ -641,7 +641,7 @@ add_and_return_legacy_wrapping_ufunc_loop(PyUFuncObject *ufunc,
         Py_DECREF(info);
         return NULL;
     }
-
+    Py_DECREF(info);  /* now borrowed from the ufunc's list of loops */
```
Comment (charris) [inline numpy/core/src/umath/dispatching.c:644]: `ufunc` now holds a new reference?

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**